### ISSUE TITLE
QT5 Threads not captured in the debugger via VSCode

### ISSUE THREAD

**Issue by d3r3kk** [Microsoft/ptvsd#428]
We have an issue reported against Microsoft/vscode-python#176 where the user reports that our debugger won't stop on a breakpoint within a QTThread process.

## Reproduction steps 
Ensure VSCode is installed/updated and the vscode-python extension is installed/updated as well. 

````
> mkdir pyQtThreadTest
> cd pyQtThreadTest
> python -m venv venv
> venv/scripts/activate
> pip install PyQt5
> code .
````
> add attached file to the project 'qt_thread_test.py'

In VSCode, ensure the latest Python extension is installed. Then, highlight line 16 in the qt_thread_test.py file and run the debugger by hitting F5.

## Expected Behaviour
The breakpoint is hit and execution stops on the breakpoint set.

## Observed Behaviour
The breakpoint is ignored and execution runs without being stopped.

**Comment by int19h** [comment]
To clarify, are they using the experimental debugger? The old one never supported this at all.

**Comment by d3r3kk** [comment]
@int19h  Yes, they are and if you do the repro you will see it fails to break on the breakpoint when using both debuggers.

**Comment by fabioz** [comment]
I tested this inside the pydevd master version (without ptvsd) and it works properly.

-- note: you need to press the button for the thread to be run in the example (this is in the original report but not here).

Not sure what was the last sync pydevd -> ptvsd... I see there's a pull request open: https://github.com/Microsoft/ptvsd/pull/426 (so, maybe just applying it could fix the issue).

**Comment by d3r3kk** [comment]
Updated instructions to be clearer, thanks. 

Will wait to see if the next version fixes the problem then.

**Comment by DonJayamanne** [comment]
Created an issue https://github.com/Microsoft/ptvsd/issues/435, I believe that should fix this issue.

**Comment by DonJayamanne** [comment]
@d3r3kk 
I've tested this at my end using the `experimental` debugger and it works as expected.
![pyqt](https://user-images.githubusercontent.com/1948812/40526690-08d53fba-5f9d-11e8-9132-1c809b1b7902.gif)

**Comment by fabioz** [comment]
@DonJayamanne no, the default is having it disabled if `--qt-support` is not passed. 

I've gotten the latest release (2018.4.0) and on Linux I can't get it to work as you're getting there, although it works if I add the code below to the start of the module:

```
from _pydev_bundle import pydev_monkey_qt
pydev_monkey_qt.patch_qt('auto')
```

(which is what passing `--qt-support=auto` would do).

So not sure why it's working on your machine... is it possible you have some local change or you're using the development version and this is already fixed there?

**Comment by Gazoo101** [comment]
Just thought I'd chime in an say that I'm seeing the same issue with using PySide2 (aka Qt for Python) with Python 3 and the latest Vs Code.

Any breakpoint I set in code executed via the overwritten run method is never hit:

```
class WorkerThread(QtCore.QThread):

   def __init__(self, workFunc, parent = None):
      QtCore.QThread.__init__(self, parent)
      self.workFunc = workFunc

   # According to the wise Qt sages - do not call run() directly.
   # Instead commence execution via the start() function

   def run(self):
      self.workFunc()
```

Even a breakpoint on the `self.workFunc()` line is never hit.

**Comment by KestutisMa** [comment]
I also can't get breakpoint hit inside QThread run() method, 1.32.3 (2019-03-14) version of vscode, linux, pyqt5. Does anybody managed to do this?

**Comment by karthiknadig** [comment]
@KestutisMa import `ptvsd` and add this to the thread function, `ptvsd.debug_this_thread()`.
https://github.com/Microsoft/ptvsd/issues/1189#issuecomment-468406399

**Comment by KestutisMa** [comment]
Thanks! ;-) probably it would be useful to add that in Python extension documentation or tutorial.
